### PR TITLE
Save all uservars

### PR DESCRIPTION
**Pull request by IgnaceBleukx**
In some edge cases, variables were unknown to the solver when constraints are simplified by our transformations.
A prominent example is the `simplify_bool` transformation which translates something like `cp.any(a,b,c) <= 10` to `True`.

I now added a "solver_vars" call before the solve call with all user variables.
It remains to be seen how this affects performance tho... But from a user perspective this is very welcome.

### CODE REVIEW

File: cpmpy/solvers/gurobi.py [hunk 1]
```diff
@@ -83,9 +83,9 @@ def __init__(self, cpm_model=None, subsolver=None):
         Arguments:
         - cpm_model: a CPMpy Model()
         """
-        if not self.supported():
-            raise Exception(
-                "CPM_gurobi: Install the python package 'gurobipy' and make sure your licence is activated!")
+        # if not self.supported():
```
Comment (Wout4) [inline cpmpy/solvers/gurobi.py:86]: We want to throw this exception right, is this still leftover from testing?

Comment (IgnaceBleukx) [reply cpmpy/solvers/gurobi.py:86]: Yes, this is leftover from testing... Licence was expired ;)

**Comment by Wout4** [comment]
This is nice, we needed it.
I added a solveAll to the tests, to check if it affects this, but seems to fail for pysdd and choco..

**Comment by IgnaceBleukx** [comment]
Aha yes indeed, nice catch.
I corrected it for Choco and added some exceptions for the other solvers.

For Pysdd I'm too unfamiliar witht the code to fix it I'm afraid

**Comment by Wout4** [comment]
looks like pysdd internally uses more literals than we give it, and that is messing up the solution count. Will try to fix

File: cpmpy/solvers/choco.py [hunk 2]
```diff
@@ -338,6 +343,7 @@ def __add__(self, cpm_expr):
         """
         # add new user vars to the set
         get_variables(cpm_expr, collect=self.user_vars)
+        # ensure all vars are known to solver
```
Comment (Dimosts) [inline cpmpy/solvers/choco.py:346]: ?

**Review by Dimosts — APPROVED** [review]
Seems good to me too. Will just remove a commented line that does not make sense in choco.

**Pull request merged**